feat(vue): add i18n routing

zanmato opened the pull request:
This adds the option to define additional routes by locale, per route. 

Route names are prefixed by `locale__` which prevents any duplicated route names. `getData` URLs are also prefixed by locale in order to prevent cache issues in case a route has the same name in multiple locales.

The user can choose to prefix their routes with the option`localePrefix` or `localeDomains` if they want to match per domain instead.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included (vue kitchensink)
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/main/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/main/CODE_OF_CONDUCT.md)

zanmato commented:
@galvez Will you have a look at this when you have some time please? I had to bring back some of the code replacing params, but perhaps there's a better way of passing these params such as `import.meta.env` etc?

galvez commented:
Will review properly and play with it tonight. Thanks!

zanmato commented:
Updated it to use env vars instead of the old virtual module inserts code

galvez commented:
Why did you revert support for env vars?

galvez commented:
Ah got it, they go through plugin options now, nice!

zanmato commented:
I'm trying it out in my production project, and this was nicer than the env variables.
However I'm missing some form of `currentLocale` whenever `localeDomains` is used

galvez commented:
Can you reach on https://discord.gg/fastify?

galvez commented:
> Running this in prod with no issues, but I'm thinking of introducing a `dataPath` to get rid of duplicate locales when running with `localePrefix` because the locale is prefixed to the path which also includes the locale.. (`/-/data/en/en/route-here`)

Would you like to address this weekend that before we put it out there? I think it's fine as-is, for a first release. But that does sound like a good idea :)

zanmato commented:
> > Running this in prod with no issues, but I'm thinking of introducing a `dataPath` to get rid of duplicate locales when running with `localePrefix` because the locale is prefixed to the path which also includes the locale.. (`/-/data/en/en/route-here`)
> 
> Would you like to address this weekend that before we put it out there? I think it's fine as-is, for a first release. But that does sound like a good idea :)

Fixed it and rebased it

zanmato commented:
We plan to allow this to be implemented by using your own virtual files, instead of bloating the vue package. This will require some new hooks to be introduced, some form of metadata per route and a key for the route map
